portal-api/src/v1/controllers/amendments/put-amendment.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { ApiError, CustomExpressRequest, PortalFacilityAmendmentUserValues } from '@ukef/dtfs2-common';
+import { generatePortalAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import api from '../../api';
+
+export type PutAmendmentRequest = CustomExpressRequest<{
+  params: {
+    facilityId: string;
+  };
+  reqBody: PortalFacilityAmendmentUserValues;
+  query: { dealId: string };
+}>;
+
+/**
+ * Upserts a draft portal facility amendment into the database
+ * @param req - The request object
+ * @param res - The response object
+ */
+export const putAmendment = async (req: PutAmendmentRequest, res: Response) => {
+  const { facilityId } = req.params;
+  const { dealId } = req.query;
+
+  try {
+    const amendmentResponse = await api.putPortalFacilityAmendment({
+      dealId,
+      facilityId,
+      amendmentUpdate: req.body,
+      auditDetails: generatePortalAuditDetails(req.user._id),
+    });
+
```

Comment:
I would suggest validating `amendmentResponse` before sending `200` what if it is falsy?

Reply:
@MarRobSoftwire and I are confident this can never be falsy at this stage (it would have thrown an error to be caught below if it was) due to how Mark has typed everything in the dtfs-central layer